### PR TITLE
Ticketing system

### PR DESCRIPTION
**Pull request by Bharani0012**
1. updated custom field table schema as per new instruction
2. delete functionality updated according to new instruction

### CODE REVIEW

File: src/backend/db/database_manager.py [hunk 1]
```diff
@@ -81,7 +82,43 @@ def add_customer(self):
...
+                updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
+            );
+            """
+            session.execute(text(create_tickets_table_query))
+
+            # Creating custom_fields table if not exists
+            create_custom_fields_table_query = """
+            CREATE TABLE IF NOT EXISTS custom_fields (
+                field_name VARCHAR(255) PRIMARY KEY,
+                field_type ENUM('VARCHAR(255)', 'INT', 'BOOLEAN', 'DATE', 'MEDIUMTEXT') NOT NULL,
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:106]: Just make field_type as 

field_type VARCHAR(255) NOT NULL

File: src/backend/db/database_manager.py [hunk 2]
```diff
@@ -259,3 +296,128 @@ def check_customer_guid_exists(self, customer_guid):
         finally:
             session.close()
 
+    #Custom Fields Related Functions
+    def add_custom_field(self, customer_guid, field_name, field_type, required):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+
+        # Define a set of allowed SQL data types
+        allowed_sql_types = {"VARCHAR(255)", "INT", "BOOLEAN", "DATE", "MEDIUMTEXT"}
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:305]: Make it allowed_custom_field_sql_types
Make it a class variable instead of a function variable

File: src/backend/db/database_manager.py [hunk 3]
```diff
@@ -259,3 +296,128 @@ def check_customer_guid_exists(self, customer_guid):
...
+                    "field_name": field_name,
+                    "field_type": field_type,
+                    "required": required,
+                },
+            )
+
+            # Dynamically add column to custom_field_values table
+            alter_table_query = f"""
+            ALTER TABLE custom_field_values
+            ADD COLUMN {field_name} {field_type};
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:330]: If there are rows present in the table, make sure the column added contains NULL values

like 

ALTER TABLE custom_field_values
            ADD COLUMN {field_name} {field_type}; with value as NULL 
            
            for all the pre-existing rows

File: src/backend/docker-compose.yml [hunk 4]
```diff
@@ -118,5 +118,4 @@ volumes:
 
 # Define the absolute path for PROJECT_ROOT
 x-environment:
-  PROJECT_ROOT: ${PROJECT_ROOT}
```
Comment (SaravanakumarR2018) [inline src/backend/docker-compose.yml:121]: Can you remove this unnecessary change

File: src/backend/db/database_manager.py (hunk 2 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:305]: updated sir

File: src/backend/db/database_manager.py (hunk 1 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:106]: updated sir

File: src/backend/db/database_manager.py (hunk 3 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:330]: sir by default it add the new column with null values

File: src/backend/docker-compose.yml (hunk 4 above)
Comment (Bharani0012) [reply src/backend/docker-compose.yml]: removed sir
